extensions/amp-consent/0.1/consent-config.js

```diff
@@ -289,6 +299,36 @@ export class ConsentConfig {
   }
 }
 
+/**
+ * Expand consent endpoint url
+ * @param {!Element} element
+ * @param {string} url
+ * @return {!Promise<string>}
+ */
+export function expandConsentEndpointUrl(element, url) {
+  return Services.urlReplacementsForDoc(element).expandUrlAsync(
+    url,
+    {
+      'CLIENT_ID': getConsentCID(element),
+    },
+    CONSENT_VARS_ALLOWED_LIST
+  );
+}
+
+/**
+ * Return AMP CONSENT scoped CID
+ * @param {!Element|!../../../src/service/ampdoc-impl.AmpDoc} node
+ * @return {!Promise<string>}
+ */
+export function getConsentCID(node) {
+  return Services.cidForDoc(node).then((cid) => {
+    return cid.get(
+      {scope: CID_SCOPE, createCookieIfNotPresent: true},
```

Comment:
Note: I fixed the scope of the CLIENT_ID API. Because I don't want the CMP to be able set any scope value to get the tracking id (`_ga` for example). 
The `CLIENT_ID` here is more like a USER_CONSENT_ID. All CMPs will share the same CID. (In the case where the publisher switch to a different CMP provider).

I could switch to a different variable name given I'm not using the exact `CLIENT_ID` API to be more specific. (e.g `USER_ID`). Let me know if you prefer that.